Registration and Login Modules

ahmedsameed opened the pull request:
First PR

ahmedsameed committed:
First Ahmed sjsjs
First Commit Ahmed

ahmedsameed committed:
new command

dheeraj837 committed:
added LoginRegistration.java and RegistrationForm.java
Resolved Registration Bug (routes are implemented between login and registration pages)

dheeraj837 committed:
Added actual Database URL, username password and
connected to MYSQL using JDBC

Yazneshpara commented on src/LoginRegisterApp.java:
```diff
@@ -0,0 +1,126 @@
+
+public class LoginRegisterApp extends JFrame {
+    // Database credentials
+    static final String DB_URL = "jdbc:mysql://localhost:3008/car_rental";
```
// Database URL (replace with your MySQL details)

Yazneshpara replied:
// Database URL (replace with your MySQL details)

Yazneshpara commented on src/LoginRegisterApp.java:
```diff
@@ -0,0 +1,126 @@
+public class LoginRegisterApp extends JFrame {
+    // Database credentials
+    static final String DB_URL = "jdbc:mysql://localhost:3008/car_rental";
+    static final String USER = "root";
```
// MySQL username

Yazneshpara commented on src/LoginRegisterApp.java:
```diff
@@ -0,0 +1,126 @@
+    // Database credentials
+    static final String DB_URL = "jdbc:mysql://localhost:3008/car_rental";
+    static final String USER = "root";
+    static final String PASS = "tiger";
```
// MySQL password

Yazneshpara commented on src/LoginRegisterApp.java:
```diff
@@ -0,0 +1,126 @@
+    // GUI components
+    private JTextField usernameField;
+    private JPasswordField passwordField;
+    private JButton loginButton, registerButton;
```
// Buttons for login and registration

Yazneshpara commented on src/LoginRegisterApp.java:
```diff
@@ -0,0 +1,126 @@
+        setTitle("Car Rental System - Login & Register");
+        setSize(400, 300);
+        setDefaultCloseOperation(JFrame.EXIT_ON_CLOSE);
+        setLayout(new BorderLayout());
```
// Use BorderLayout for window layout

Yazneshpara commented on src/LoginRegisterApp.java:
```diff
@@ -0,0 +1,126 @@
+
+        // Creating UI elements
+        JLabel titleLabel = new JLabel("Login", JLabel.CENTER);
+        titleLabel.setFont(new Font("Arial", Font.BOLD, 18));
```
// Set the font style for the title

Yazneshpara commented on src/LoginRegisterApp.java:
```diff
@@ -0,0 +1,126 @@
+        titleLabel.setFont(new Font("Arial", Font.BOLD, 18));
+
+        JPanel inputPanel = new JPanel();
+        inputPanel.setLayout(new GridLayout(3, 2));
```
// GridLayout with 3 rows and 2 columns

Yazneshpara commented on src/LoginRegisterApp.java:
```diff
@@ -0,0 +1,126 @@
+        JLabel usernameLabel = new JLabel("Username:");
+        usernameField = new JTextField();
+        JLabel passwordLabel = new JLabel("Password:");
+        passwordField = new JPasswordField();
```
// Password field for hidden input

Yazneshpara commented on src/LoginRegisterApp.java:
```diff
@@ -0,0 +1,126 @@
+        inputPanel.setLayout(new GridLayout(3, 2));
+
+        JLabel usernameLabel = new JLabel("Username:");
+        usernameField = new JTextField();
```
// Text field to enter username

Yazneshpara commented on src/LoginRegisterApp.java:
```diff
@@ -0,0 +1,126 @@
+        inputPanel.add(usernameField);
+        inputPanel.add(passwordLabel);
+        inputPanel.add(passwordField);
+
```
// Creating login and register buttons

Yazneshpara commented on src/RegistrationForm.java:
```diff
@@ -0,0 +1,92 @@
+import javax.swing.*;
```
// For creating GUI elements

Yazneshpara commented on src/RegistrationForm.java:
```diff
@@ -0,0 +1,92 @@
+import java.awt.*;
+import java.awt.event.ActionEvent;
+import java.awt.event.ActionListener;
+import java.sql.*;
```
// For connecting and interacting with a MySQL database

Yazneshpara commented on src/RegistrationForm.java:
```diff
@@ -0,0 +1,92 @@
+import java.awt.event.ActionEvent;
+import java.awt.event.ActionListener;
+import java.sql.*;
+
```
// Class for creating a registration form for a new user

Yazneshpara commented on src/RegistrationForm.java:
```diff
@@ -0,0 +1,92 @@
+    static final String DB_URL = "jdbc:mysql://localhost:3306/car_rental_system";
+    static final String USER = "root";  
+    static final String PASS = "password";
+
```
// Constructor for creating the registration form

Yazneshpara commented on src/RegistrationForm.java:
```diff
@@ -0,0 +1,92 @@
+
+        setVisible(true);
+    }
+
```
// Method to register a user in the database

dheeraj837 committed:
Added jar files and dependencies for mySQL connection

ahmedsameed reviewed (commented):
Please add jar files for SQL server also as part of dependencies. Use DB URL, USERNAME and PASSWORD as environment variables. Thank you.

dheeraj837 committed:
Add files via upload

dheeraj837 committed:
Updated Login page by adding HomePage path/url

dheeraj837 committed:
added home page which contain >rent a car. >return a car. options

dheeraj837 committed:
This page displays available cars for renting

dheeraj837 committed:
Update RentCarPage and add new windows for Booking and Confirmation

dheeraj837 committed:
Merge branch 'firstbranch' of https://github.com/ahmedsameed/Car-Rental-System into firstbranch

ahmedsameed committed:
Deleted AvailableCarspage.java

ahmedsameed committed:
Deleted rentcarpage1.java

dheeraj837 committed:
Your commit message

Yazneshpara committed:
yaznesh

dheeraj837 committed:
Resolved Calling of RentCar

dheeraj837 committed:
added booking window to select the date,
removed timeslots

dheeraj837 committed:
added dynamic pricing- backend functionality

dheeraj837 committed:
truncated the price to 2 decimal place

dheeraj837 committed:
Removed unwanted comments

dheeraj837 committed:
implemented user story 3- added logic for displaying alternative cars when selected car is already booked

ahmedsameed committed:
For issue, As a user I want to be displayed the alternative cars in the order of car that are most suitable to my selected car first. Type casting was redone so that function displayAlternativeCars is taking Car parameter properly
selectCar parameter is set when car date is fetched to check for
availability. Removed the code iterating over every car to create
alternativecars window to optimize space and processing time

ahmedsameed committed:
For issue, As a user I want to be displayed the alternative cars in the order of car that are most suitable to my selected car first. Type casting was redone so that function displayAlternativeCars is taking Car parameter properly selectCar parameter is set when car date is fetched to check for availability. Removed the code iterating over every car to create alternativecars window to optimize space and processing time

dheeraj837 committed:
1.Resolved issue with NULL POINTER EXCEPTION
rising out of date_added fields on BookingWindow.java page
2.Added intermediary window diaplying the customer a message of car unavailabitlity
Creating a  frame Car unavailabitlity notification. Added alternative car button with
event listener.

Yazneshpara committed:
made implementation of issue 15 where i developed the return a car feature by adding new class and constructor

Yazneshpara committed:
Created a page where rented car can be returned

dheeraj837 committed:
added unit testing for carReturnWindow with 5 test cases
